src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package mate.academy.service.impl;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+
+    private static final String HASHING_ALGORITHM = "SHA-256";
+    @Inject
+    private UserService userService;
+
+    public AuthenticationServiceImpl(UserService userService) {
+        this.userService = userService;
+    }
```

Comment:
The constructor is initializing `userService`, which might conflict with the `@Inject` annotation. If using a DI framework, rely on it to inject dependencies and remove this constructor.